### ISSUE TITLE
Keras implementation - The convergence is not as good as in the numpy implementation

### ISSUE THREAD

**Issue by mfournigault** [mfournigault/inverse_compositional_algorithm#18]
- The convergence is not as good as in the numpy implementation. The gradient dp converges much slower than in the reference results. 
For example, taking the first unit test with the rubber whale image and the translation, 22 iterations are necessary to converge on the first scale with the keras implementation, where only 5 are necessary to converge with the numpy implementation. This might be linked to the first issue.
- See logs in verbose mode in the notebook test_ica_keras.ipynb pushed in the commit 1ba91cdef14641349b14edbc7a9221bdd83718d4 and compare with those of the notebook inverse_compositional_algorithm_robust.ipynb
